File: HashTable.go
package main

import (
	"fmt"
)

type HashTable struct {
	data [][][]string
}

func NewHashTable(size int) *HashTable {
	return &HashTable{
		data: make([][][]string, size),
	}
}

func (h *HashTable) _hash(key string) int {
	hash := 0
	for i := 0; i < len(key); i++ {
		hash = (hash + int(key[i])*i) % len(h.data)
	}
	return hash
}

func (h *HashTable) set(key string, value string) [][][]string {
	address := h._hash(key)
	if h.data[address] == nil {
		h.data[address] = make([][]string, 0)
	}
	h.data[address] = append(h.data[address], []string{key, value})
	return h.data
}

func (h *HashTable) get(key string) string {
	address := h._hash(key)
	currentBucket := h.data[address]
	if currentBucket != nil {
		for i := 0; i < len(currentBucket); i++ {
			if currentBucket[i][0] == key {
				return currentBucket[i][1]
			}
		}
	}
	return ""
}

func main() {
	myHashTable := NewHashTable(100)
	myHashTable.set("082124606606", "Rony Setyawan")
	fmt.Println(myHashTable.get("082124606606"))
}

File: HashTable.kt
class HashTable(size: Int) {
    private val data = arrayOfNulls<Array<Pair<String, Any>>>(size)

    private fun hash(key: String): Int {
        var hash = 0
        for (i in key.indices) {
            hash = (hash + key[i].code * i) % data.size
        }
        return hash
    }

    operator fun set(key: String, value: Any) {
        val address = hash(key)
        if (data[address] == null) {
            data[address] = emptyArray()
        }
        data[address] = data[address]!! + arrayOf(key to value)
    }

    operator fun get(key: String): Any? {
        val address = hash(key)
        val currentBucket = data[address]
        if (currentBucket != null) {
            for (i in currentBucket.indices) {
                if (currentBucket[i].first == key) {
                    return currentBucket[i].second
                }
            }
        }
        return null
    }
}

fun main(){
    val myHashTable = HashTable(100)
    myHashTable["082124606606"] = "Rony Setyawan"
    println(myHashTable["082124606606"])
}
